src/main/java/com/hcl/appscan/jenkins/plugin/scanModes/PostmanCollection.java

```diff
@@ -0,0 +1,146 @@
+package com.hcl.appscan.jenkins.plugin.scanModes;
+
+import com.hcl.appscan.jenkins.plugin.Messages;
+import hudson.Extension;
+import hudson.Util;
+import hudson.util.FormValidation;
+import hudson.util.VariableResolver;
+import org.jenkinsci.Symbol;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.QueryParameter;
+
+import java.util.Map;
+
+public class PostmanCollection extends ScanMode {
+
+    private static final String POSTMAN_COLLECTION = "Postman Collection"; //$NON-NLS-1$
+
+    private String m_postmanCollectionFile;
+    private String m_additionalDomains;
+    private String m_environmentalVariablesFile;
+    private String m_globalVariablesFile;
+    private String m_additionalFiles;
+
+    @DataBoundConstructor
+    public PostmanCollection() {
+        this("", "", "", "", "");
+    }
+
+    public PostmanCollection(String postmanCollectionFile, String additionalDomains, String environmentalVariablesFile, String globalVariablesFile, String additionaFiles) {
+        m_postmanCollectionFile = postmanCollectionFile;
+        m_additionalDomains = additionalDomains;
+        m_environmentalVariablesFile = environmentalVariablesFile;
+        m_globalVariablesFile = globalVariablesFile;
+        m_additionalFiles = additionaFiles;
+    }
+
+    @DataBoundSetter
+    public void setPostmanCollectionFile(String postmanCollectionFile) {
+        m_postmanCollectionFile = postmanCollectionFile;
+    }
+
+    public String getPostmanCollectionFile() {
+        return m_postmanCollectionFile;
+    }
+
+    @DataBoundSetter
+    public void setAdditionalDomains(String additionalDomains) {
+        m_additionalDomains = additionalDomains;
+    }
+
+    public String getAdditionalDomains() {
+        return m_additionalDomains;
+    }
+
+    @DataBoundSetter
+    public void setEnvironmentalVariablesFile(String environmentalVariablesFile) {
+        m_environmentalVariablesFile = environmentalVariablesFile;
+    }
+
+    public String getEnvironmentalVariablesFile() {
+        return m_environmentalVariablesFile;
+    }
+
+    @DataBoundSetter
+    public void setGlobalVariablesFile(String globalVariablesFile) {
+        m_globalVariablesFile = globalVariablesFile;
+    }
+
+    public String getGlobalVariablesFile() {
+        return m_globalVariablesFile;
+    }
+
+    @DataBoundSetter
+    public void setAdditionalFiles(String additionalFiles) {
+        m_additionalFiles = additionalFiles;
+    }
+
+    public String getAdditionalFiles() {
+        return m_additionalFiles;
+    }
+
+
+    @Override
+    public Map<String, String> configureScanProperties(Map<String, String> properties, VariableResolver<String> resolver) {
```

Comment:
The 5 conditionals in this method all do essentially the same thing, just working on different member variables. This logic should be pulled out into a private function that can be called 5 times, rather than duplicating the code 5 times.